app/demo-geant-integration/HitRootIO.hh

```diff
@@ -0,0 +1,87 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file demo-geant-integration/HitRootIO.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <G4ThreadLocalSingleton.hh>
+#include <G4Types.hh>
+#include <G4VHit.hh>
+
+class TFile;
+class TTree;
+class TBranch;
+class G4Event;
+
+namespace demo_geant
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Example of output event structure stored in ROOT.
+ */
+class HitRootEvent
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using HitContainer = std::map<std::string, std::vector<G4VHit*>>;
+    //!@}
+
+  public:
+    HitRootEvent() = default;
+
+    void SetEventID(G4int id) { event_id_ = id; }
+    HitContainer* GetHCMap() { return &hcmap_; }
+
+  private:
+    G4int event_id_{0};
```

Comment:
In all other classes we've been using `int`, `bool` (I see a `G4bool` in `HitRootIO` as well)... these seem to be a legacy standard, so should we just move basic C++ types instead of the G4 typedefs for consistency with the rest of the code?

Reply:
Yeah we had a discussion about this... it seems to be "the geant way" but it will never be any different from the raw types. I think if we want this to be more of an "app" (driver for testing coupled problems) than a "demo" (reference for Geant4 users) we should just use int/bool. But then again most of the geant interfaces do specify the inputs in terms of `G4int`... @drbenmorgan is there any plan to officially deprecate those aliases? Maybe leave `G4Types.hh` but change all the interfaces and internals to use standard types.

Reply:
Okay using int/bool/std::string everywhere instead of Geant4 types for now.